run_unittests.py

```diff
@@ -484,6 +499,40 @@ def assertPathBasenameEqual(self, path, basename):
         path_basename = PurePath(path).parts[-1]
         self.assertEqual(PurePath(path_basename), PurePath(basename), msg)
 
+    def assertBuildIsNoop(self):
+        ret = self.build()
+        if self.backend is Backend.ninja:
+            self.assertEqual(ret.split('\n')[-2], self.no_rebuild_stdout)
+        elif self.backend is Backend.vs:
+            # Ensure that some target said that no rebuild was done
+            self.assertIn('CustomBuild:\n  All outputs are up-to-date.', ret)
+            self.assertIn('ClCompile:\n  All outputs are up-to-date.', ret)
+            self.assertIn('Link:\n  All outputs are up-to-date.', ret)
+            # Ensure that no targets were built
+            clre = re.compile('ClCompile:\n [^\n]*cl', flags=re.IGNORECASE)
+            linkre = re.compile('Link:\n [^\n]*link', flags=re.IGNORECASE)
+            self.assertNotRegex(ret, clre)
+            self.assertNotRegex(ret, linkre)
+        elif self.backend is Backend.xcode:
+            raise unittest.SkipTest('Please help us fix this test on the xcode backend')
+        else:
+            raise RuntimeError('Invalid backend: {!r}'.format(self.backend.name))
+
+    def assertRebuiltTarget(self, target):
+        ret = self.build()
+        if self.backend is Backend.ninja:
+            self.assertIn('Linking target {}'.format(target), ret)
+        elif self.backend is Backend.vs:
+            # Ensure that this target was rebuilt
+            clre = re.compile('ClCompile:\n [^\n]*cl[^\n]*' + target, flags=re.IGNORECASE)
+            linkre = re.compile('Link:\n [^\n]*link[^\n]*' + target, flags=re.IGNORECASE)
```

Comment:
__[flake8]__
* [F841] local variable 'linkre' is assigned to but never used

<!-- Comment by [SideCI](https://sideci.com) -->